src/crafting.cpp

```diff
@@ -688,14 +688,48 @@ comp_selection<item_comp> player::select_item_component( const std::vector<item_
         uimenu cmenu;
         // Populate options with the names of the items
         for( auto &map_ha : map_has ) {
-            std::string tmpStr = item::nname( map_ha.type ) + _( " (nearby)" );
+            std::string tmpStr;
+            if( item::count_by_charges( map_ha.type ) ) {
+                tmpStr = string_format( _( "%s (%d/%d nearby)" ),
+                                        item::nname( map_ha.type ),
+                                        map_ha.count,
+                                        map_inv.charges_of( map_ha.type ) );
+            } else {
+                tmpStr = string_format( _( "%s (%d/%d nearby)" ),
+                                        item::nname( map_ha.type ),
+                                        map_ha.count,
+                                        map_inv.amount_of( map_ha.type ) );
+            }
             cmenu.addentry( tmpStr );
         }
         for( auto &player_ha : player_has ) {
-            cmenu.addentry( item::nname( player_ha.type ) );
+            std::string tmpStr;
```

Comment:
There is a staggering amount of code duplication here. This is basically the same as above, and it's repeated again below. And both branches of the `if` look very similar as well. This should probably be extracted into some function.

Reply:
Not 100% sure of what I'd be extracting ot a function but have reduced the duplication using a conditional in the string_format call instead of an almost identical if block.